examples/ct/ct_handlers_test.go

```diff
@@ -358,12 +372,42 @@ func TestAddChain(t *testing.T) {
 	assert.Equal(t, "BAEABnNpZ25lZA==", resp.Signature)
 }
 
-// Submit a chain with a valid precert but not complete path to root. Should be rejected.
-func TestAddPrecertChainMissingIntermediate(t *testing.T) {
+// Submit a chain with a valid precert but not signed by next cert in chain. Should be rejected.
+func TestAddPrecertChainInvalidPath(t *testing.T) {
+	km := new(crypto.MockKeyManager)
+	client := new(trillian.MockTrillianLogClient)
 
+	roots := loadCertsIntoPoolOrDie(t, []string{testonly.CACertPEM})
+	reqHandlers := CTRequestHandlers{0x42, roots, client, km, time.Millisecond * 500, fakeTimeSource}
+
+	cert, err := fixchain.CertificateFromPEM(testonly.PrecertPEMValid)
+	_, ok := err.(x509.NonFatalErrors)
+
+	if err != nil && !ok {
+		t.Fatal(err)
+	}
+
+	pool := NewPEMCertPool()
```

Comment:
Why the 'pool' name?  Are you memoizing certs once seen and storing a set of unique certs, or something?